### DISCUSSION TITLE
Display results of Action inside Modal?

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#3108] [Help] [answered]
I need to open a Modal with some dependent select fields,
upon selection of the final one a url should appear in the modal for the user to copy.
(Url is built from values of related records, selected in the dropdowns)

For this to work, I would have to somehow display the results of the action, inside the (still opened) Modal...

Is that somehow doable with a standard `Action`? Or how else could I go about that?

**Comment by danharrin** [comment, marked as answer]
Maybe use a disabled text input underneath the selects? Then use `$set` to set the input to the generated URL, using `afterStateUpdated()` on each select

**Reply by madsem** [reply]
And how could I do this inside a Modal? Sorry if my question is obvious haha, I kind of have scatterbrain atm and don't quite know where/how to start with this. :D

<img width="1239" alt="Screen Shot 2022-07-14 at 12 28 40 PM" src="https://user-images.githubusercontent.com/3166991/178962310-79261bdd-fa7a-4072-abe9-911f6cb5260a.png">

The problem I am facing is this:

A `Lander` model can have many `Domain` models, so ideally the user should be able to click an action link inside the table.
Then select options, among these options is then also the domains.

Then based on these selections query the relationships, build the url with the correct parameters in  the query string and display it.

And right now I'm completely unsure if I can even use an `Action` for this scenario, or how to implement this with an alternative method.

**Reply by danharrin** [reply]
```php
Action::make('trackingLink')
    ->form([
         Select::make('first_select')
              ->options(...blabla options)
              ->reactive()
              ->afterStateUpdated(function ($get, $set) {
                  $first = $get('first_select');
                  $second = $get('second_select');

                  $set('url', "https://domain.com/{$first}/{$second}");
              }),
          Select::make('second_select')
              ->options(...blabla options)
              ->reactive()
              ->afterStateUpdated(function ($get, $set) {
                  $first = $get('first_select');
                  $second = $get('second_select');

                  $set('url', "https://domain.com/{$first}/{$second}");
              }),
         TextInput::make('url')->disabled(),
    ])
    ->modalActions(fn ($action): array => [
        $action->getModalCancelAction()->label('Close')
    ])
    ->action(function () {})
```

**Reply by madsem** [reply]
Oh wow, thanks @danharrin you are a rockstar!
Let me check if this gives the desired result, but I have a starting point now. Thanks again

**Reply by madsem** [reply]
Yes, this is going to work somehow. Amazing.

This was the secret sauce:
```php
->modalActions(fn($action): array => [
                    $action->getModalCancelAction()->label('Close'),
                ])
                ->action(function () {
                })
```

And that I can use `$get` and `$set` in the way you showed. Super cool!
